### PR TITLE
Allow 2d input if RNN order is BatchLastIndex

### PR DESCRIPTION
**Pull request by NeroBlackstone**
issue #767

### CODE REVIEW

**Comment by NeroBlackstone** [comment]
Forgot remove RNN error check test, will fix it tomorrow

**Comment by avik-pal** [comment]
We should add a check comparing the results for a matrix input and a 3d Array input with a single batch dimension, to verify that the current impl is doing the right thing.

File: test/layers/recurrent_tests.jl [hunk 1]
```diff
@@ -340,6 +338,16 @@ end
                         @test size(y) == (5,)
                         @test length(y_) == 4
                         @test all(x -> size(x) == (5,), y_)
+                        
+                        if x isa AbstractMatrix
```
Comment (avik-pal) [inline test/layers/recurrent_tests.jl:342]:
```suggestion
                        if x isa AbstractMatrix && ordering isa BatchLastIndex
```

File: test/layers/recurrent_tests.jl [hunk 2]
```diff
@@ -340,6 +338,16 @@ end
...
                         @test all(x -> size(x) == (5,), y_)
+                        
+                        if x isa AbstractMatrix && ordering isa BatchLastIndex
+                            x2 = reshape(x, Val(3))
+                            
+                            y2,_= rnn(x2,ps,st)
+                            @test y == vec(y2)
+                            
+                            y2_,_= rnn_seq(x2, ps, st)
+                            @test all(x -> x[1] == vec(x[2]), zip(y_,y2_))
```
Comment (avik-pal) [inline test/layers/recurrent_tests.jl:349]:
```suggestion
                            y2, _ = rnn(x2,ps,st)
                            @test y == vec(y2)
                            
                            y2_, _ = rnn_seq(x2, ps, st)
                            @test all(x -> x[1] == vec(x[2]), zip(y_,y2_))
```

File: test/layers/recurrent_tests.jl [hunk 3]
```diff
@@ -340,6 +338,16 @@ end
...
+                        
+                        if x isa AbstractMatrix && ordering isa BatchLastIndex
+                            x2 = reshape(x, Val(3))
+                            
+                            y2, _ = rnn(x2,ps,st)
+                            @test y == vec(y2)
+                            
+                            y2_, _ = rnn_seq(x2, ps, st)
+                            @test all(x -> x[1] == vec(x[2]), zip(y_,y2_))
+                        end
```
Comment (avik-pal) [inline test/layers/recurrent_tests.jl:350]:
```suggestion
                        if x isa AbstractMatrix && ordering isa BatchLastIndex
                            x2 = reshape(x, Val(3))
                            
                            y2, _ = rnn(x2, ps, st)
                            @test y == vec(y2)
                            
                            y2_, _ = rnn_seq(x2, ps, st)
                            @test all(x -> x[1] == vec(x[2]), zip(y_, y2_))
                        end
```

**Review by avik-pal — APPROVED** [review]

**Pull request merged**